### PR TITLE
feat(Interactions): add subcommand parsing section

### LINKED ISSUE #754: Include a subsection on subcommands.
**Problem**
Currently, subcommands are only mentioned in [registering-slash-commandsl#option-types](https://discordjs.guide/interactions/registering-slash-commands.html#option-types) and you are referred to the Discord API for further information. This is fine, however, you have to go searching the documentation for how to find which subcommand a user opted to select.

**Prefered Solution**
In [replying-to-slash-commands#parsing-options](https://discordjs.guide/interactions/replying-to-slash-commands.html#parsing-options) there should be a short section explaining how to get the subcommand (or subcommandGroup) via `.getSubcommand()` or `.getSubcommandGroup()`

**Alternatives**
Keep it as is and allow people to navigate to the [`CommandInteractionOptionResolver`](https://discord.js.org/#/docs/main/master/class/CommandInteractionOptionResolver) on the documentation and find the relevant methods there.

**Additional Notes**
I would be happy to provide a PR if this is a change that you would like to make.

### PR DESCRIPTION
**Pull request by adrifcastr**
**Please describe the changes this PR makes and why it should be merged:**
Closes #754 and splits the parsing options section into regular option parsing and a detailed example on parsing subcommands and their options.

### CODE REVIEW

File: guide/interactions/replying-to-slash-commands.md [hunk 1]
```diff
@@ -278,6 +280,59 @@ console.log([string, integer, boolean, user, member, channel, role, mentionable]
 If you want the Snowflake of a structure instead, grab the option via `get()` and access the Snowflake via the `value` property. Note that you should use `const { value: name } = ...` here to [destructure and rename](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) the value obtained from the <DocsLink path="typedef/CommandInteractionOption">`CommandInteractionOption`</DocsLink> structure to avoid identifier name conflicts.
 :::
 
+### Subcommands
+
+If you have a command that contains subcommands you can parse them in a very similar way as to the above examples.
```
Comment (DaStormer) [inline guide/interactions/replying-to-slash-commands.md:285]:
```suggestion
If you have a command that contains subcommands, you can parse them in a very similar way as to the above examples.
```
just need a comma here

File: guide/interactions/replying-to-slash-commands.md [hunk 2]
```diff
@@ -278,6 +280,59 @@ console.log([string, integer, boolean, user, member, channel, role, mentionable]
...
+```
+
+The following snippet details the logic needed to parse the subcommands and respond accordingly using the `CommandInteractionOptionResolver#getSubcommand()` method:
+
+```js {5-20}
+client.on('interactionCreate', async interaction => {
+	if (!interaction.isCommand()) return;
+
+	if (interaction.commandName === 'info') {
+		if (interaction.options.getSubcommand() === 'user') {
```
Comment (NotSugden) [inline guide/interactions/replying-to-slash-commands.md:321]:
```suggestion
      const subCommand = interaction.options.getSubCommand();
		if (subCommand === 'user') {
```
and below use the `subCommand` variable

**Review by NotSugden — CHANGES_REQUESTED** [review]

File: guide/interactions/replying-to-slash-commands.md (hunk 2 above)
Comment (adrifcastr) [reply guide/interactions/replying-to-slash-commands.md:321]: unsure if that's a cleaner approach, imo the direct comparison on the resolver method is clear and concise

Comment (DaStormer) [reply guide/interactions/replying-to-slash-commands.md:321]: this suggestion is incorrect - it's `.getSubcommand()`, not `.getSubCommand()`

**Pull request merged**